RATapi/inputs.py

```diff
@@ -151,11 +151,23 @@ def make_problem(project: RATapi.Project) -> ProblemDefinition:
         The problem input used in the compiled RAT code.
 
     """
-    hydrate_id = {"bulk in": 1, "bulk out": 2}
     prior_id = {"uniform": 1, "gaussian": 2, "jeffreys": 3}
 
-    # Ensure backgrounds and resolutions have a source defined
+    # Ensure all contrast fields are properly defined
     for contrast in project.contrasts:
+        contrast_fields = ["data", "background", "bulk_in", "bulk_out", "scalefactor", "resolution"]
+
+        if project.calculation == Calculations.Domains:
+            contrast_fields.append("domain_ratio")
+
+        for field in contrast_fields:
+            if getattr(contrast, field) == "":
+                raise ValueError(
+                    f'In the input project, the {field} of contrast "{contrast.name}" does not have a '
```

Comment:
`f'In the input project, the "{field}" of contrast "{contrast.name}" does not have a '`